tools/docker/kratos_image/Dockerfile

```diff
@@ -0,0 +1,65 @@
+# Copyright 2018- The Pixie Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+FROM --platform=$BUILDPLATFORM golang:1.24-alpine@sha256:9fadeb603e14f1f3e08bdbec6681fa14446053c498a554f3e57260bf892c487e AS build
+
+ARG TARGETOS TARGETARCH
+ARG KRATOS_VERSION
+
+ARG GOOS=$TARGETOS
+ARG GOARCH=$TARGETARCH
+
+WORKDIR /src
+
+RUN apk update
+RUN apk add git
+
+RUN git clone --depth 1 https://github.com/ory/kratos.git
+WORKDIR /src/kratos
+RUN git fetch --tags && git checkout $KRATOS_VERSION
+
+# The hydra and kratos migrations attempt to create the same table.
+# Kratos's migration only creates the table while hydra's migration creates the table
+# and inserts a row. Therefore, we replace kratos's problematic version with empty placeholders
+# to avoid a conflict. Removing the problematic migration file prior to module initialization and
+# compilation did not work.
+RUN for file in persistence/sql/migrations/sql/20150100000001000000_networks.*.sql; do echo "-- Migration disabled to avoid conflict with hydra" > "$file"; done
```

Comment:
this might be riskier than just migrations, if both services are writing to the same tables then we might end up with bad data.
see https://github.com/ory/hydra/issues/3391#issuecomment-1377489023

do you want to try using different schemas in the database? you can update the psql uri to indicate the schema by including the `options=--search_path` param (though you probably need to create the schema externally).